File: DSA/VedantChouhan_590024563/Day-88/q2.c
#include <stdio.h>

int buyChoco(int prices[], int pricesSize, int money) {
    int min1 = 1001;
    int min2 = 1001;

    for (int i = 0; i < pricesSize; i++) {
        if (prices[i] < min1) {
            min2 = min1;
            min1 = prices[i];
        }
        else if (prices[i] < min2) {
            min2 = prices[i];
        }
    }

    int cost = min1 + min2;

    if (cost <= money)
        return money - cost;

    return money;
}

int main() {
    int n, money;

    printf("Enter number of chocolates: ");
    scanf("%d", &n);

    int prices[n];

    printf("Enter chocolate prices: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &prices[i]);
    }

    printf("Enter money: ");
    scanf("%d", &money);

    printf("Remaining money: %d\n",
           buyChoco(prices, n, money));

    return 0;
}

File: DSA/vaishnavi_joshi_590016947/Day89/Q2.java
class Solution {
    public int buyChoco(int[] prices, int money) {
        
        int min1 = Integer.MAX_VALUE;
        int min2 = Integer.MAX_VALUE;
        
        for (int price : prices) {
            
            if (price < min1) {
                min2 = min1;
                min1 = price;
            } 
            else if (price < min2) {
                min2 = price;
            }
        }
        
        int totalCost = min1 + min2;
        
        // If we can afford both chocolates
        if (totalCost <= money) {
            return money - totalCost;
        }
        
        // Otherwise, money remains unchanged
        return money;
    }
}